### ISSUE TITLE
"Start debugging Item" in Xamarin.Android project, if RELEASE is chosen, doesn't launch app

### ISSUE THREAD

**Issue by knocte** [mono/monodevelop#4190]
Steps to reproduce:
1. Load a solution in VS4Mac 7.4.build1033 that contains a Xamarin.Android app.
2. Switch the build configuration from DEBUG to RELEASE.
3. Right click on Xamarin.Android project -> Start debugging Item.

Current results:
Starts compiling the app, deploys it, and when it launches it, it just stops (the "stop" icon becomes a "play" icon).

Expected results:
Should probably tell the developer to switch back from RELEASE to DEBUG first, instead of (silently) failing.

> VS bug [#587465](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/587465)

**Comment by sgmunn** [comment]
Can you please attach IDE logs, Android log files as well please.

**Comment by knocte** [comment]
I've tried to reproduce this again with the version of VS4Mac(7.4.3 build 10) that I have now (to get the logs), and now I get some interesting log messages in the "Deploying to Device" panel, which I was not getting before:

```
Done building project "GWallet.Frontend.XamForms.Droid.fsproj".

Build succeeded.

/Library/Frameworks/Mono.framework/Versions/5.8.1/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(2057,5): warning MSB3277: Found conflicts between different versions of "System.Runtime" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
/Library/Frameworks/Mono.framework/Versions/5.8.1/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(2057,5): warning MSB3277: Found conflicts between different versions of "Microsoft.CSharp" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
/Library/Frameworks/Mono.framework/Versions/5.8.1/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(2057,5): warning MSB3277: Found conflicts between different versions of "System.Net.Http" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
/Users/andres/Documents/Code/gwalletFRONTEND/src/GWallet.Frontend.XamForms.Droid/GWallet.Frontend.XamForms.Droid.fsproj : warning XA0105: The $(TargetFrameworkVersion) for FormsViewGroup.dll (vv8.0) is greater than the $(TargetFrameworkVersion) for your project (v7.1). You need to increase the $(TargetFrameworkVersion) for your project.
/Users/andres/Documents/Code/gwalletFRONTEND/src/GWallet.Frontend.XamForms.Droid/GWallet.Frontend.XamForms.Droid.fsproj : warning XA0105: The $(TargetFrameworkVersion) for Xamarin.Forms.Platform.Android.dll (vv8.0) is greater than the $(TargetFrameworkVersion) for your project (v7.1). You need to increase the $(TargetFrameworkVersion) for your project.
/Users/andres/Documents/Code/gwalletFRONTEND/src/GWallet.Frontend.XamForms.Droid/GWallet.Frontend.XamForms.Droid.fsproj : warning XA0105: The $(TargetFrameworkVersion) for Xamarin.Forms.Platform.dll (vv8.0) is greater than the $(TargetFrameworkVersion) for your project (v7.1). You need to increase the $(TargetFrameworkVersion) for your project.
/Users/andres/Documents/Code/gwalletFRONTEND/src/GWallet.Frontend.XamForms.Droid/GWallet.Frontend.XamForms.Droid.fsproj : warning XA0105: The $(TargetFrameworkVersion) for ZXing.Net.Mobile.Forms.Android.dll (vv8.0) is greater than the $(TargetFrameworkVersion) for your project (v7.1). You need to increase the $(TargetFrameworkVersion) for your project.
EXEC : warning : Directory obj/Release/android/assets contains FSharp.Core.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains FastAndroidCamera.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Java.Interop.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Microsoft.CSharp.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Mono.Security.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains NBitcoin.Litecoin.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains NBitcoin.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.ABI.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.Accounts.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.Contracts.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.Hex.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.JsonRpc.Client.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.JsonRpc.RpcClient.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.KeyStore.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.RLP.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.RPC.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.Signer.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.StandardTokenEIP20.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.Util.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Nethereum.Web3.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Newtonsoft.Json.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains SharpRavenLight.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Core.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Globalization.Extensions.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Net.Http.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Numerics.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Runtime.Serialization.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.ServiceModel.Internals.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Xml.Linq.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.Xml.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains System.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.FSharp.ResourceProvider.Runtime.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Animated.Vector.Drawable.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Annotations.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Compat.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Core.UI.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Core.Utils.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Design.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Fragment.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Media.Compat.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Transition.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.Vector.Drawable.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.v4.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.v7.AppCompat.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.v7.CardView.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.v7.MediaRouter.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.v7.Palette.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Android.Support.v7.RecyclerView.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Xamarin.Forms.Platform.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains ZXing.Net.Mobile.Core.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains ZXing.Net.Mobile.Forms.Android.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains ZXing.Net.Mobile.Forms.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains ZXingNetMobile.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains Zlib.Portable.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains mscorlib.dll but no debug symbols file was found.
EXEC : warning : Directory obj/Release/android/assets contains zxing.portable.dll but no debug symbols file was found.
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(479,2): error : Failure [INSTALL_FAILED_NO_MATCHING_ABIS: Failed to extract native libraries, res=-113]
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(479,2): error :   at Mono.AndroidTools.Internal.AdbOutputParsing.CheckInstallSuccess (System.String output, System.String packageName) [0x0011f] in /Users/builder/data/lanes/5809/a7829590/source/monodroid/tools/msbuild/external/androidtools/Mono.AndroidTools/Internal/AdbOutputParsing.cs:333 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(479,2): error :   at Mono.AndroidTools.AndroidDevice+<>c__DisplayClass94_0.<InstallPackage>b__0 (System.Threading.Tasks.Task`1[TResult] t) [0x00016] in /Users/builder/data/lanes/5809/a7829590/source/monodroid/tools/msbuild/external/androidtools/Mono.AndroidTools/AndroidDevice.cs:746 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(479,2): error :   at System.Threading.Tasks.ContinuationTaskFromResultTask`1[TAntecedentResult].InnerInvoke () [0x00024] in <e22c1963d07746cd9708456620d50e1a>:0 
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(479,2): error :   at System.Threading.Tasks.Task.Execute () [0x00010] in /Users/builder/data/lanes/5533/mono-mac-sdk/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Task.cs:2502 
    63 Warning(s)
    1 Error(s)

Time Elapsed 00:00:10.96
```

This only happens when RELEASE is selected. When DEBUG is selected, deployment+debugging works fine. Is it enough or should I also provide log files?

**Comment by knocte** [comment]
Log files in the 7.0 directory:

[7.0.zip](https://github.com/mono/monodevelop/files/1927144/7.0.zip)

**Comment by sgmunn** [comment]
> Failure [INSTALL_FAILED_NO_MATCHING_ABIS: Failed to extract native libraries, res=-113]

this is the issue, the ABIs supported by the app are not compatible with the device. You need to include the correct ABI in the project options and we need to do a better job of making that more obvious.

**Comment by knocte** [comment]
> this is the issue, the ABIs supported by the app are not compatible with the device.

If this is the issue, then why does debugging in DEBUG work, and why does running a RELEASE build of my apk on a device work? Why is RELEASE binaries an issue with the emulator? I don't understand...

**Comment by sgmunn** [comment]
The ABIs that are produced for each configuration are different. By default, debug includes x86 which is probably the ABI of the emulator whereas the selected ABIs for release include only arm by default (which is the most likely ABI for a device.

**Comment by knocte** [comment]
Ah, makes sense.

**Comment by knocte** [comment]
Why was it closed? https://devdiv.visualstudio.com/DevDiv/_workitems/edit/587465 gives me a 401

**Comment by mrward** [comment]
@knocte It seems it was closed because it has been fixed in VS Mac 7.6.0.1801 (release-7.6 branch)

**Comment by knocte** [comment]
@mrward thanks
